homeassistant/components/braviatv/coordinator.py

```diff
@@ -256,3 +257,16 @@ async def async_send_command(self, command: Iterable[str], repeats: int) -> None
         for _ in range(repeats):
             for cmd in command:
                 await self.client.send_command(cmd)
+
+    @catch_braviatv_errors
+    async def async_learn_command(self, entity_id: str) -> None:
+        """Display a list of available commands in a persistent notification."""
+        commands = await self.client.get_command_list()
+        codes = ", ".join(commands.keys())
+        title = "Bravia TV"
+        message = f"**List of available commands for `{entity_id}`**:\n\n{codes}"
+        persistent_notification.async_create(
```

Comment:
Where are the commands learnt?

Reply:
I'm sorry, I didn't quite understand the question. List of commands are learned by the user.

Reply:
This service is meant to allow the device to learn, ie record, new commands. It's not meant to show the available commands to the user.

Reply:
> This service is meant to allow the device to learn, ie record, new commands.

The name of the service is confusing. I think in this case it should have been called `remote.add_command` or `remote.record_command`. [The documentation says](https://developers.home-assistant.io/docs/core/entity/remote/#learn-command) "Learn a command from a device" and "Send commands to a device".

Reply:
Yes, please update the docs to use a better wording if you have time. We can't change the name of the service though.

Will you revert this PR?

Reply:
> Will you revert this PR?

What's the alternative? I would like to leave it.

Reply:
I don't think there's any alternative. The service in this PR doesn't follow our intended design.

Reply:
Very sad. Is it allowed to create a separate service call like `androidtv.learn_sendevent`?

Reply:
Could it be a custom service ie `braviatv.show_available_remote_commands`? Or the button entity can trigger such a notification with a command list.

Reply:
We also can add new service call `remote.print_commands` to Remote Entity. I think it will be useful to find out what commands are recorded/supported.

Reply:
I think this should just be part of the integration docs. We use the persistent notification rarely and only when the user needs to take action. We don't use it for general information about the integration.

Reply:
The commands are different for different TV models. It would be problematic to list them all in the docs.
Revert #77306

Reply:
So maybe the best option would be a custom service that prints the supported commands to the log?

Reply:
Perhaps there is an alternative solution: if we send an unknown command, we can give an error with a list of available commands. However, the user will still have to go into the logs.

Reply:
Logging errors and available commands is good and standard behavior.

Reply:
Done in #77329